web/src/data/recoil.ts

```diff
@@ -110,12 +122,41 @@ export const vaporConnectionConfig = selector<ConnectionConfig | undefined>({
   },
 });
 
+export const portalConnectionConfig = selector<ConnectionConfig | undefined>({
+  key: "portalConnectionConfig",
+  get: async ({ get }) => {
+    const portalHostnameValue = get(portalHostname);
+    const portalCredentialsValue = get(portalCredentials);
+
+    if (portalCredentialsValue) {
+      let decodedCredentials;
+      try {
+        decodedCredentials = atob(portalCredentialsValue);
+      } catch (e) {
+        console.error("error conecting to Portal", { e });
+      }
+      if (decodedCredentials) {
+        const { username, password } = JSON.parse(decodedCredentials);
+        return {
```

Comment:
check that username and password are not undefined or empty (if they are, the credentials were not shaped correctly)